application-development.md

```diff
@@ -19,25 +19,43 @@ The kubelet executes the check and decides if the container should be restarted.
 - The official Kubernetes documentation offers some practical advice on how to [configure Liveness, Readiness and Startup Probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/).
 - [Liveness probes are dangerous](https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html) has some information on how to set (or not) dependencies in your readiness probes.
 
-### Containers have readiness probes
+### Containers have Readiness probes
 
 > Please note that there's no default value for readiness and liveness.
 
 If you don't set the readiness probe, the kubelet assumes that the app is ready to receive traffic as soon as the container starts.
 
 If the container takes 2 minutes to start, all the requests to it will fail for those 2 minutes.
 
-### Avoid using the Liveness probe (if you can)
+### Containers crash when there's a fatal error
 
-The Liveness probe is designed to restart your container when it's stuck due to edge-cases such as a dead-lock.
+If the application reaches an unrecoverable error, [you should let it crash](https://blog.colinbreck.com/kubernetes-liveness-and-readiness-probes-revisited-how-to-avoid-shooting-yourself-in-the-other-foot/#letitcrash).
 
-_But if you can detect the deadlock and signal a failing liveness probe, why not exiting from the deadlock in the first place?_
+Examples of such unrecoverable errors are:
 
-It's a fair question, and that's perhaps the reason why liveness probes are generally not necessary.
+- an uncaught exception
+- a typo in the code (for dynamic languages)
+- unable to load a header or dependency
 
-It's a best practice to let the app crash rather than signalling a failing liveness probe.
+Please note that you should not signal a failing Liveness probe.
 
-You can find [more details about using (or not) liveness probes in this article](https://blog.colinbreck.com/kubernetes-liveness-and-readiness-probes-revisited-how-to-avoid-shooting-yourself-in-the-other-foot/#letitcrash).
+Instead, you should immediately exit the process and let the kubelet restart the container.
+
+### Configure a passive Liveness probe
+
+The Liveness probe is designed to restart your container when it's stuck.
+
+Consider the following scenario: if your application is processing an infinite loop, there's no way to exit or ask for help.
+
+When the process is consuming 100% CPU, it won't have time to reply to the Readiness probe checks, and it will be eventually removed from the Service.
```

Comment:
Should it say "it won't have time to reply to the liveness probe checks,...?

Reply:
Readiness is actually what I meant. You have ghost pod because it's running, but not part of the Service. So it consumes one of the slots in the deployment without contributing.

I think I should amend the description and make it more clear.